src/commands/cmd_list.cc

```diff
@@ -555,6 +555,101 @@ class CommandLMove : public Commander {
   bool dst_left_;
 };
 
+class CommandLPOS : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    target_ = args[2];
+    CommandParser parser(args, 3);
+    while (parser.Good()) {
+      if (parser.EatEqICase("RANK")) {
+        rank_ = GET_OR_RET(parser.TakeInt());
+        if (rank_ == 0) {
+          return {Status::RedisExecErr, "RANK can't be zero"};
+        }
+      } else if (parser.EatEqICase("COUNT")) {
+        count_ = GET_OR_RET(parser.TakeInt());
+        if (count_ < 0) {
+          return {Status::RedisExecErr, "COUNT can't be negative"};
+        }
+      } else if (parser.EatEqICase("MAXLEN")) {
+        max_len_ = GET_OR_RET(parser.TakeInt());
+        if (max_len_ < 0) {
+          return {Status::RedisExecErr, "MAXLEN can't be negative"};
+        }
+      } else {
+        return parser.InvalidSyntax();
+      }
+    }
+    return Commander::Parse(args);
+  }
+
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    redis::List list_db(svr->storage, conn->GetNamespace());
+    // get length of the list
+    uint64_t nums = 0;
+    auto s = list_db.Size(args_[1], &nums);
+    if (!s.ok() && !s.IsNotFound()) {
+      return {Status::RedisExecErr, s.ToString()};
+    }
+    // rank_ can be positive ot negative. if it's zero,
+    // it means the user doesn't specify it.
+    if (rank_ == 0) rank_ = 1;
+    // it means the user doesn't specify it.
+    if (count_ == -1) count_ = 1;
+    // it means the user doesn't specify it.
+    if (max_len_ == -1) max_len_ = 0;
```

Comment:
When user doesn't specify it, would you mind handle it in `Parse`, or just set default to these values?

Reply:
I will use default value. Sorry